packages/data-api/src/validators.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { hasNoContent, takesDateForm, takesIdForm, yup, yupUtils } from '@tupaia/utils';
+
+export const eventOptionsValidator = yup.object().shape({
+  organisationUnitCodes: yup.array().of(yup.string().required()).strict().required(),
+  startDate: yup
+    .string()
+    .test(
+      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'startDate should be in ISO 8601 format'),
+    ),
+  endDate: yup
+    .string()
+    .test(
+      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'endDate should be in ISO 8601 format'),
+    ),
+  dataElementCodes: yup.array().of(yup.string().required()).strict(),
+  dataGroupCode: yup.string().strict().required(),
```

Comment:
Needing to use `strict` here really surprised me. It turns out that by default `yup` will attempt to cast its input to the schema time (ie. `1` would be cast to `'1'` and then accepted by this validator).

I found this pretty unintuitive, and definitely gave me a bit of discomfort as I've probably not realised it was needed for other validators when using `yup`.

Overall, I'm a bit on the fence about `yup`. It's great to have an OOTB package to handle validation for us, and it plays fairly well with typescript. But this input modification, it's poor support for dynamic types, and its syntax aren't the best imo...

Reply:
Oh wow, that is surprising! There's no way I would have expected that. I think it's even worth highlighting at dev chat, as it's pretty widely used but I don't know how widely this quirk would be known

Reply:
Good point! I'll add it to the agenda!